### PR TITLE
Leaderboard db

### PR DESCRIPTION
**Pull request by Edmund-Lui98**
leaderboard tab now pulls data from the db to show cards

### CODE REVIEW

File: lib/ballerCard.dart [hunk 1]
```diff
@@ -103,22 +95,40 @@ class LeaderBallerCard extends StatelessWidget {
...
-          Expanded(
-            child: Icon(MyFlutterApp.basketball_ball))
+          Expanded(child: Icon(MyFlutterApp.basketball_ball))
         ],
       ),
     );
   }
 }
+
+class User {
```
Comment (hoklaamc) [inline lib/ballerCard.dart:122]: Can we move this class to a new file? `/lib/models/user.dart` would be good 👍

File: lib/pages/login_signup_page.dart [hunk 2]
```diff
@@ -1,342 +1,342 @@
-import 'package:flutter/material.dart';
-import 'package:hoopr/services/authentication.dart';
+// import 'package:flutter/material.dart';
```
Comment (hoklaamc) [inline lib/pages/login_signup_page.dart:1]: What happened here? :o

File: lib/services/db.dart [hunk 3]
```diff
@@ -0,0 +1,36 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
```
Comment (hoklaamc) [inline lib/services/db.dart:1]: Let's avoid abbreviations and change this to `database.dart` - it's not too long

File: lib/services/authentication.dart [hunk 4]
```diff
@@ -1,11 +1,12 @@
 import 'dart:async';
 import 'package:firebase_auth/firebase_auth.dart';
-import 'package:hoopr/services/user.dart';
+import 'package:hoopr/services/db.dart';
```
Comment (hoklaamc) [inline lib/services/authentication.dart:3]: Update this to `package:hoopr/services/database.dart` if we change that file name

File: lib/services/db.dart [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import 'package:hoopr/ballerCard.dart';
```
Comment (hoklaamc) [inline lib/services/db.dart:2]: Update this import to be `package:hoopr/models/user.dart` once that class is moved

File: lib/ballerCard.dart [hunk 6]
```diff
@@ -103,22 +95,40 @@ class LeaderBallerCard extends StatelessWidget {
...
+  const User({
+    this.firstName,
+    this.lastName,
+    this.username,
+    this.bp,
```
Comment (hoklaamc) [inline lib/ballerCard.dart:127]: Does `bp` stand for `ballerPoints`? Can we just have `points` here for clarity and simplicity sake?

File: lib/ballerCard.dart [hunk 7]
```diff
@@ -103,22 +95,40 @@ class LeaderBallerCard extends StatelessWidget {
...
+  });
+
+  final String firstName;
+  final String lastName;
+  final String username;
+  final int bp;
```
Comment (hoklaamc) [inline lib/ballerCard.dart:133]: Related to above `bp` -> `points` comment

File: lib/services/db.dart [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  final CollectionReference userCollection =
+      Firestore.instance.collection('users');
+
+  Future updateUserData(
+      String firstName, String lastName, String username, int bp) async {
+    return await userCollection.document(uid).setData({
+      'firstName': firstName,
+      'lastName': lastName,
+      'username': username,
+      'bp': bp
```
Comment (hoklaamc) [inline lib/services/db.dart:18]: Related to above `bp` -> `points` comment

File: lib/services/db.dart [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+    });
+  }
+
+  List<User> _userListFromSnapshot(QuerySnapshot snapshot) {
+    return snapshot.documents.map((doc) {
+      return User(
+        firstName: doc.data["firstName"],
+        lastName: doc.data["lastName"],
+        username: doc.data["username"],
+        bp: doc.data["bp"],
```
Comment (hoklaamc) [inline lib/services/db.dart:28]: Related to above `bp` -> `points` comment

File: lib/user_list.dart [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+  @override
+  _UserListState createState() => _UserListState();
+}
+
+class _UserListState extends State<UserList> {
+  @override
+  Widget build(BuildContext context) {
+    final users = Provider.of<List<User>>(context);
+
+    print(users.length);
```
Comment (hoklaamc) [inline lib/user_list.dart:15]: Remove print

File: lib/user_list.dart [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+
+    return ListView.builder(
+        itemCount: users.length,
+        itemBuilder: (context, index) {
+          return LeaderBallerCard(
+              bp: users[index].bp,
```
Comment (hoklaamc) [inline lib/user_list.dart:21]: Related to above `bp` -> `points` comment

File: lib/user_list.dart [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+              name: users[index].username,
+              pic: AssetImage('assets/raysmall.png'));
```
Comment (hoklaamc) [inline lib/user_list.dart:23]: Can we rename this variable to be something like `profileImage`? Not sure where this is defined

**Review by hoklaamc — CHANGES_REQUESTED** [review]
Looks good if everything is working properly :) However ;et's make some changes to clean up the code to make it easier to maintain

File: lib/ballerCard.dart (hunk 1 above)
Comment (Edmund-Lui98) [reply lib/ballerCard.dart:122]: Yes, once we finish the MVP we can clean up the code

File: lib/pages/login_signup_page.dart (hunk 2 above)
Comment (Edmund-Lui98) [reply lib/pages/login_signup_page.dart:1]: This page isn't being used but throws errors due to other file changes so I commented them out for now. We can delete this file when cleaning up the code.

File: lib/services/db.dart (hunk 3 above)
Comment (Edmund-Lui98) [reply lib/services/db.dart:1]: Yes definitely. I was working on an old branch and there was another database.dart so I used db.dart for now.

**Review by hoklaamc — APPROVED** [review]

**Pull request merged**